$OpenBSD: patch-libraries_Cabal_Cabal_Distribution_Simple_Compiler_hs,v 1.1 2018/01/22 00:42:30 kili Exp $

Dynamic libraries are still disabled on OpenBSD, so don't let Cabal
do anything about them (especially putting a bogus dynamic-library-dirs
field into package registrations).

Index: libraries/Cabal/Cabal/Distribution/Simple/Compiler.hs
--- libraries/Cabal/Cabal/Distribution/Simple/Compiler.hs.orig
+++ libraries/Cabal/Cabal/Distribution/Simple/Compiler.hs
@@ -329,15 +329,7 @@ backpackSupported = ghcSupported "Support Backpack"
 -- | Does this compiler support a package database entry with:
 -- "dynamic-library-dirs"?
 libraryDynDirSupported :: Compiler -> Bool
-libraryDynDirSupported comp = case compilerFlavor comp of
-  GHC ->
-      -- Not just v >= mkVersion [8,0,1,20161022], as there
-      -- are many GHC 8.1 nightlies which don't support this.
-    ((v >= mkVersion [8,0,1,20161022] && v < mkVersion [8,1]) ||
-      v >= mkVersion [8,1,20161021])
-  _   -> False
- where
-  v = compilerVersion comp
+libraryDynDirSupported comp = False
 
 -- | Does this compiler support Haskell program coverage?
 coverageSupported :: Compiler -> Bool
